add list users, delete users for idps

mystiberry-okta opened the pull request:
Are the new operations in the right place?
Anything missing? 
Any extra explanations or gotchas needed?

mystiberry-okta commented:
@tthompson-okta and anyone at @federations-okta who's interested, please review.

mystiberry-okta commented on _source/_docs/api/resources/idps.md:
```diff
@@ -3816,6 +3816,145 @@ curl -v -X POST \
+### Find Users
+{:.api .api-operation} 
+
+<span class="api-uri-template api-uri-get"><span class="api-label">GET</span> /api/v1/idps/*:id*/users</span>
```
Oops! Will do!

tthompson-okta commented on _source/_docs/api/resources/idps.md:
```diff
@@ -3816,6 +3816,145 @@ curl -v -X POST \
+    "lastUpdated": "2016-06-05T00:50:39.000Z",
+    "profile": {
+      "lastName": "Kim",
+      "subjectNameQualifier": null,
```
Hrmm, this looks like a SAML IDP example (given all the subjectName stuff in the profile). It's not a huge deal, but would be nice to use a Social IDP user profile for the example, as that's slightly more human readable (the attributes are not cryptic things like subjectSpProvidedId). Here's an example you can use:

```json
{
    "id": "00u5cl9lo7nMjHjPr0h7",
    "externalId": "109912936038778",
    "created": "2015-11-03T19:10:11.000Z",
    "lastUpdated": "2015-11-03T19:11:49.000Z",
    "profile": {
        "firstName": "Carol",
        "middleName": "Lee",
        "lastName": "Johnson",
        "email": "carol_johnson@tfbnw.net",
        "displayName": "Carol Johnson",
        "profile": "https://www.facebook.com/app_scoped_user_id/109912936038778/"
    },
    "_links": {
      "self": {
        "href": "https://example.okta.com/api/v1/idps/0oa4lb6lbtmH355Hx0h7/users/00u5cl9lo7nMjHjPr0h7",
        "hints": {
            "allow": [
                "GET",
                "DELETE"
              ]
        }
      },
      "idp": {
          "href": "https://example.okta.com/api/v1/idps/0oa4lb6lbtmH355Hx0h7"
      },
      "user": {
          "href": "https://example.okta.com/api/v1/users/00u5cl9lo7nMjHjPr0h7"
      }
   }
}
```

mystiberry-okta replied:
Done!

tthompson-okta commented on _source/_docs/api/resources/idps.md:
```diff
@@ -3816,6 +3816,145 @@ curl -v -X POST \
+]
+~~~
+
+### Delete Users
```
This operation should be titled `Unlink User From IdP`

mystiberry-okta commented:
@tthompson-okta  I addressed your comments (fixed the syntax code for all the operations, not just my new ones). You might want to double-check the request and response examples to make sure I got the IDs right for idps and user. Should anyone else review this?

mystiberry-okta commented:
@tthompson-okta  Need a second review.

tthompson-okta commented on _source/_docs/api/resources/idps.md:
```diff
@@ -3816,12 +3816,108 @@ curl -v -X POST \
+### Find Users
+{:.api .api-operation} 
+
+{% api_operation get /api/v1/idps/*:id*/users %}
```
Everywhere else we omit the `api/v1` part of the path; should probably make it consistent here. I'd guess we did that so that the docs don't automatically become dated if/when we decide to ship /api/v2.

mystiberry-okta replied:
Removed. We should show the whole string, and use text variables for the inevitable changes. I can do this easily in XML/DITA, not sure how to do it in kramdown.

tthompson-okta commented on _source/_docs/api/resources/idps.md:
```diff
@@ -3816,12 +3816,108 @@ curl -v -X POST \
+
+{% api_operation get /api/v1/idps/*:id*/users %}
+
+Find all the users linked to an identify provider
```
Type: should be `identity provider`

mystiberry-okta replied:
Fixed.

tthompson-okta commented on _source/_docs/api/resources/idps.md:
```diff
@@ -3816,12 +3816,108 @@ curl -v -X POST \
+##### Response Parameters
+{:.api .api-response .api-response-params}
+
+List of the org's [identity providers](#identity-provider-model) whether activated or not
```
This seems like a copy/paste error? The response is not a list of identity providers, it's a list of linked IdP users. I would just say `List of the users that are linked to the specified identity provider`.

mystiberry-okta replied:
done.

mystiberry-okta commented on _source/_docs/api/resources/idps.md:
```diff
@@ -3816,12 +3816,108 @@ curl -v -X POST \
+
+{% api_operation delete /api/v1/idps/*:id*/users/*:uid* %}
+
+Delete the specified user from the IdP
```
Sorry I missed updating the description when I changed the title.

mystiberry-okta replied:
Done.

mystiberry-okta commented:
@tthompson-okta Ship it?

tthompson-okta commented:
🚢
